### PR TITLE
Added userscript to open feed links.

### PR DESCRIPTION
**Pull request by jnphilipp**
Add a userscript to open feed links from tags <link rel="related" type="application/rss+xml" href="..."> of a page.

### CODE REVIEW

**Comment by The-Compiler** [comment]
I'm a bit opposed to parsing HTML via `grep` I have to say :wink:

Would you mind if I close this and add an userscript using `BeautifulSoup` (a Python HTML parser) instead?

**Comment by jnphilipp** [comment]
No, not really.

File: misc/userscripts/openfeeds [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#
+# Use the hotkey to open the feeds in new tab/window, press 'gF' to open
+#
+
+import os
+import re
+
+from bs4 import BeautifulSoup
+
+if os.environ['QUTE_HTML']:
```
Comment (The-Compiler) [inline misc/userscripts/openfeeds]: I don't see why `QUTE_HTML` should ever be empty - was it for you?

File: misc/userscripts/openfeeds [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    soup = BeautifulSoup(open(os.environ['QUTE_HTML'], 'r'))
```
Comment (The-Compiler) [inline misc/userscripts/openfeeds]: You should also close the `QUTE_HTML` file properly:

```python
with open(os.environ['QUTE_HTML'], 'r') as f:
    soup = BeautifulSoup(f)
```

File: misc/userscripts/openfeeds [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    with open(os.environ['QUTE_FIFO'], 'w') as f:
+        for link in soup.find_all('link', rel='alternate', type=re.compile(r'application/((rss|rdf|atom)\+)?xml|text/xml')):
+            f.write('open -t %s\n' % link.get('href'))
+    f.close()
```
Comment (The-Compiler) [inline misc/userscripts/openfeeds]: No need for this - the `with` block will close it for you.

**Comment by jnphilipp** [comment]
No, it wasn't. I tought of maybe adding a part to call it so and then starting qutebrowser with the new tabs.

**Comment by The-Compiler** [comment]
Ah - then you'd need to do something like `'QUTE_HTML` in os.environ` instead, as you'd get a `KeyError` when it's not set at all.

**Comment by jnphilipp** [comment]
Yeah, right.

**Comment by jnphilipp** [comment]
Rewrote it with your suggested changes.

**Pull request merged**